SalesforceCommerceCloud/pwa-kit
# [App Extensibility ⚙️] Windows Readiness @@W-17273481@@

# Description

This PR contains any changes needed to get extensibility and windows tests working. This includes fixes to resolution for extensibility and general SDK changes.

As a result of this ticket, I have create the following gus issues to fix windows tests that I would not be able to fix in the timeline of this ticket, so we are going to isolate them.

https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000026KG63YAG/view
https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000026KKpLYAW/view

# Changes

- Fix extensibility path manipulation to be compatible with windows
- Fix build scripts to handle spaces in paths on windows.

# How to Test-Drive This PR

- (In Windows)
- Test that the `template-typescript-minimal` project runs.
- Create an override for the `extension-sample`'s `withRedBorder` component to ensure that overrides is working.
- Now generate a project sing this command `node .\packages\pwa-kit-create-app\scripts\create-mobify-app-dev.js --outputDir generated-project`
- Ensure the project is generated without error.
- Now open the project and update the extensions config to look like this:
```
"extensions": [
        [
          "@salesforce/extension-chakra-storefront", 
          {
            "enabled": true,
            "pages": {},
            "url": {
              "site": "path",
              "locale": "path",
              "showDefaults": true,
              "interpretPlusSignAsSpace": false
            },
            "defaultSite": "RefArchGlobal",
            "siteAliases": {
                "RefArch": "us",
                "RefArchGlobal": "global"
            },
            "sites": [
              {
                  "id": "RefArch",
                  "l10n": {
                      "supportedCurrencies": ["USD"],
                      "defaultCurrency": "USD",
                      "defaultLocale": "en-US",
                      "supportedLocales": [
                          {
                              "id": "en-US",
                              "preferredCurrency": "USD"
                          },
                          {
                              "id": "en-CA",
                              "preferredCurrency": "USD"
                          }
                      ]
                  }
              },
              {
                  "id": "RefArchGlobal",
                  "l10n": {
                      "supportedCurrencies": ["GBP", "EUR", "CNY", "JPY"],
                      "defaultCurrency": "GBP",
                      "supportedLocales": [
                          {
                              "id": "de-DE",
                              "preferredCurrency": "EUR"
                          },
                          {
                              "id": "en-GB",
                              "preferredCurrency": "GBP"
                          },
                          {
                              "id": "es-MX",
                              "preferredCurrency": "MXN"
                          },
                          {
                              "id": "fr-FR",
                              "preferredCurrency": "EUR"
                          },
                          {
                              "id": "it-IT",
                              "preferredCurrency": "EUR"
                          },
                          {
                              "id": "ja-JP",
                              "preferredCurrency": "JPY"
                          },
                          {
                              "id": "ko-KR",
                              "preferredCurrency": "KRW"
                          },
                          {
                              "id": "pt-BR",
                              "preferredCurrency": "BRL"
                          },
                          {
                              "id": "zh-CN",
                              "preferredCurrency": "CNY"
                          },
                          {
                              "id": "zh-TW",
                              "preferredCurrency": "TWD"
                          }
                      ],
                      "defaultLocale": "en-GB"
                  }
              }
            ],
            "commerceAPI": {
              "proxyPath": "/mobify/proxy/api",
              "parameters": {
                  "clientId": "c9c45bfd-0ed3-4aa2-9971-40f88962b836",
                  "organizationId": "f_ecom_zzrf_001",
                  "shortCode": "8o7m175y",
                  "siteId": "RefArchGlobal"
              }
            },
            "einsteinAPI": {
              "host": "https://api.cquotient.com/",
              "einsteinId": "1ea06c6e-c936-4324-bcf0-fada93f83bb1",
              "siteId": "aaij-MobileFirst",
              "isProduction": false
            }
          }
        ]
      ]
```
- Start the project and make sure it runs (note: the home page is going to look weird as the base project has a homepage definition that is interfering with the homepage of the extension, but the rest of the site should work)
- 
# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
